daemon/api_test.go

```diff
@@ -1151,3 +1153,147 @@ func (s *apiSuite) TestInstallLicensed(c *check.C) {
 	err := inst.dispatch()()
 	c.Check(err, check.IsNil)
 }
+
+func (s *apiSuite) TestGetCapabilities(c *check.C) {
+	d := newTestDaemon()
+	d.capRepo.Add(&caps.Capability{"serial-port", "A serial port", caps.FileType, nil})
+	req, err := http.NewRequest("GET", "/1.0/capabilities", nil)
+	c.Assert(err, check.IsNil)
+	rec := httptest.NewRecorder()
+	capabilitiesCmd.GET(capabilitiesCmd, req).ServeHTTP(rec, req)
+	c.Check(rec.Code, check.Equals, 200)
+	c.Check(rec.Body.String(), testutil.Contains, "serial-port")
+	c.Check(rec.Body.String(), check.Equals,
+		`{"result":[{"name":"serial-port","label":"A serial port",`+
+			`"type_name":"file","attrs":null}],`+
+			`"status":"OK","status_code":200,"type":"sync"}`)
+}
+
+func (s *apiSuite) TestCreateCapabilitiesGood(c *check.C) {
+	// Setup
+	d := newTestDaemon()
+	cap := &caps.Capability{
+		Name:  "name",
+		Label: "label",
+		Type:  caps.FileType,
+	}
+	capRepr := cap.ConvertToRepr()
+	text, err := json.Marshal(capRepr)
+	c.Assert(err, check.IsNil)
+	buf := bytes.NewBuffer(text)
+	// Sanity check, let's see how the representation looks like
+	c.Assert(buf.String(), check.Equals,
+		`{"name":"name","label":"label","type_name":"file","attrs":null}`)
+	// Execute
+	req, err := http.NewRequest("POST", "/1.0/capabilities", buf)
+	c.Assert(err, check.IsNil)
+	rsp := createCapability(capabilitiesCmd, req).Self(nil, nil).(*resp)
+	// Verify (external)
+	c.Check(rsp.Type, check.Equals, ResponseTypeSync)
+	c.Check(rsp.Status, check.Equals, http.StatusOK)
+	c.Check(rsp.Result, check.IsNil)
+	// Verify (internal)
+	c.Check(d.capRepo.All(), testutil.DeepContains, *cap)
```

Comment:
Instead of using All here and below, can we get an API on the capabilities repository to ask for a capability by name? Feels much nicer and easier to debug when there are issues too. Would be fine to see this in a follow up, though.